packages/components/toggle-button/src/toggle-button.ts

```diff
@@ -0,0 +1,128 @@
+import { type Icon } from '@sl-design-system/icon';
+import { type EventEmitter, EventsController, event } from '@sl-design-system/shared';
+import { type SlToggleEvent } from '@sl-design-system/shared/events.js';
+import { type CSSResultGroup, LitElement, PropertyValues, type TemplateResult, html } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './toggle-button.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-toggle-button': ToggleButton;
+  }
+}
+
+export type ToggleButtonSize = 'sm' | 'md' | 'lg';
+export type ToggleButtonFill = 'ghost' | 'outline';
+
+/**
+ * Let the user toggle between two states.
+ *
+ * ```html
+ *  <sl-toggle-button>
+ *    <sl-icon name="far-gear"></sl-icon>
+      <sl-icon name="fas-gear" slot="pressed"></sl-icon>
+    </sl-toggle-button>
+ * ```
+ *
+ * @slot default - The icon shown in the default state of the button
+ * @slot pressed - The icon shown in the pressed state of the button
+ */
+export class ToggleButton extends LitElement {
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  // eslint-disable-next-line no-unused-private-class-members
+  #events = new EventsController(this, {
+    click: this.#onClick,
+    keydown: this.#onKeydown
+  });
+
+  /** @internal The slotted default icon. */
+  @queryAssignedElements() defaultIcon?: Icon[];
+
+  /** @internal The slotted pressed state icon. */
+  @queryAssignedElements({ slot: 'pressed' }) pressedIcon?: Icon[];
+
+  /** @internal Emits when the button item has been toggled. */
+  @event({ name: 'sl-toggle' }) toggleEvent!: EventEmitter<SlToggleEvent<boolean>>;
+
+  /** Whether the toggle-button is disabled; when set no interaction is possible. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** The variant of the toggle-button. */
+  @property({ reflect: true }) fill: ToggleButtonFill = 'ghost';
```

Comment:
I still think this should be the default (`undefined`). What happens if somebody does `<sl-toggle-button fill="Outline">` in HTML?

Reply:
I know your code matches how we do things in Button, but this kinda falls under "continued insight" :)